File: 98. Validate Binary Search Tree.py
class Solution:
    def isValidBST(self, root: Optional[TreeNode]) -> bool:
        def isValid(root: TreeNode | None, min: TreeNode | None, max: TreeNode | None) -> bool:
            if not root:
                return True
            if min and root.val <= min:
                return False
            if max and root.val >= max:
                return False
            return isValid(root.left, min, root) and isValid(root.right, root, max)
        
        return isValid(root, None, None)

File: 98. Validate Binary Search Tree.go
func isValidBST(root *TreeNode) bool {
	isValid := func(root, min,)
    isValid := func(root, min, max *TreeNode) bool{
		if root == nil{return true}
		if min != nil && root.Val <= min.Val{return false}
		if max != nul && root.Val >= max.Val{return false}
		return isValid(root.left, min, root) && isValid(root.right, root, max)
	}
	return isvalid(root, nil, nil)
}